src/odemis/acq/drift/test/drift_test.py

```diff
@@ -27,14 +29,82 @@
+            elif child.name == CONFIG_SCANNER["name"]:
+                cls.scanner = child
+
+    def test_acquire(self):
+        # the acquired region should follow min_resolution and max_pixels settings
+        self.region = (0, 0, 0.1, 0.1)
+        self.dwellTime = 5e-6
+        ac = AnchoredEstimator(self.scanner, self.detector, self.region, self.dwellTime)
+        ac.acquire()
+        acquired_data = ac.raw[0]
+        acquired_res = acquired_data.shape
+        total_pixels = numpy.prod(acquired_res)
+        self.assertGreaterEqual(acquired_res, MIN_RESOLUTION)
+        self.assertLessEqual(total_pixels, MAX_PIXELS)
+
+    def test_estimate(self):
+        # maximum drift estimates should be within the FOV of the scanner
+        self.region = (0, 0, 0.1, 0.1)
+        self.dwellTime = 5e-6
+        ac = AnchoredEstimator(self.scanner, self.detector, self.region, self.dwellTime)
+        calculated_drift = ac.estimate()
+        # Compute margin so that it's always possible to place the ROI within the
+        # scanner FoV: half the ROI resolution (at scale 1).
+        margin = ((ac._res[0] * ac._scale[0]) // 2,
+                  (ac._res[1] * ac._scale[1]) // 2)
+        trans_rng = self.scanner.translation.range  # pixels
+        self._trans_range = ((trans_rng[0][0] + margin[0], trans_rng[0][1] + margin[1]),
+                             (trans_rng[1][0] - margin[0], trans_rng[1][1] - margin[1]))
+        translation = (self._trans_range[1][0] - self._trans_range[0][0],
+                       self._trans_range[1][1] - self._trans_range[0][1])
+        # assuming the x y origin is in top left corner of the image
+        # if not this case will produce junk values
+        self.assertLessEqual(calculated_drift, translation)
+
+    def test_updateSEMSettings(self):
+        # the arbitrary update value should be registered and updated when the update function is called
+        self.region = (0, 0, 0.1, 0.1)
+        self.dwellTime = 5e-6
+        ac = AnchoredEstimator(self.scanner, self.detector, self.region, self.dwellTime)
+        ac._scale = (1, 1)
+        ac._res = (1022, 1022)
+        ac._dwell_time = 2.5e-6
+        ac._updateSEMSettings()
+        # always in this order # TODO check the order of update settings
+        self.assertEqual(ac._emitter.scale.value, ac._scale)
+        self.assertEqual(ac._emitter.resolution.value, ac._res)
+        self.assertEqual(ac._emitter.dwellTime.value, ac._dwell_time)
+        # TODO how to check translation update?
+
     def test_estimateAcquisitionTime(self):
-        # TODO
-        pass
+        # min amount of acquired time should be more than 0
+        self.region = (0, 0, 0.1, 0.1)
+        self.dwellTime = 5e-6
+        ac = AnchoredEstimator(self.scanner, self.detector, self.region, self.dwellTime)
+        calculate_estimateAcquisitionTime = ac.estimateAcquisitionTime()
+        self.assertGreater(calculate_estimateAcquisitionTime, 0)
```

Comment:
would it be possible to create a bit more detailed test. Could you for instance determine the amount of pixels based on the region, then multiply that with the dwell time, to get an estimate?